packages/react-transform/test/node/index.test.tsx

```diff
@@ -0,0 +1,835 @@
+import { transform } from "@babel/core";
+import signalsTransform, { PluginOptions } from "../../src/index";
+
+function dedent(str: string) {
+	let result = str;
+
+	const lines = str.split("\n");
+	let minIndent: number = Number.MAX_SAFE_INTEGER;
+	lines.forEach(function (l) {
+		const m = l.match(/^(\s+)\S+/);
+		if (m) {
+			const indent = m[1].length;
+			if (!minIndent) {
+				// this is the first indented line
+				minIndent = indent;
+			} else {
+				minIndent = Math.min(minIndent, indent);
+			}
+		}
+	});
+
+	if (minIndent !== null) {
+		result = lines
+			.map(function (l) {
+				return l[0] === " " || l[0] === "\t" ? l.slice(minIndent) : l;
+			})
+			.join("\n");
+	}
+
+	return result.trim();
+}
+
+const toSpaces = (str: string) => str.replace(/\t/g, "  ");
+
+function transformCode(code: string, options?: PluginOptions) {
+	const signalsPluginConfig: any[] = [signalsTransform];
+	if (options) {
+		signalsPluginConfig.push(options);
+	}
+
+	const result = transform(code, {
+		plugins: [signalsPluginConfig, "@babel/plugin-syntax-jsx"],
+	});
+
+	return result?.code || "";
+}
+
+function runTest(
+	input: string,
+	expected: string,
+	options: PluginOptions = { mode: "auto" }
+) {
+	const output = transformCode(input, options);
+	expect(toSpaces(output)).to.equal(toSpaces(dedent(expected)));
+}
+
+describe("React Signals Babel Transform", () => {
+	describe("auto mode transformations", () => {
+		it("wraps arrow function component with return statement in try/finally", () => {
+			const inputCode = `
+				const MyComponent = () => {
+					signal.value;
+					return <div>Hello World</div>;
+				};
+			`;
+
+			const expectedOutput = `
+				import { useSignals as _useSignals } from "@preact/signals-react/runtime";
+				const MyComponent = () => {
+					var _effect = _useSignals();
+					try {
+						signal.value;
+						return <div>Hello World</div>;
+					} finally {
+						_effect.f();
```

Comment:
Any opinions on the name of this method `f`? It's short for `finishEffect` to stop the current effect.

I've changed the return of `useSignals` from an method to an object that can hang other stuff of like the dispose symbol for using with `using` which this PR also includes support for.